### ISSUE TITLE
TypeError: Cannot read property 'name' of undefined

### ISSUE THREAD

**Issue by albertosramires** [ceolter/ag-grid#1291]
Followed the guide at: https://www.ag-grid.com/javascript-grid-cell-rendering/

Keep getting the error in the title, if I remove `cellRendererFramework: NameCellComponent` everything runs just fine.

My cell:
```
import { Component } from "@angular/core";

import { AgRendererComponent } from "ag-grid-ng2/main";

@Component({
  selector: "name-cell",
  templateUrl: "./name.component.html"
})

export class NameCellComponent implements AgRendererComponent {
  private params: any;
  private value: string;

  agInit(params: any): void {
    this.params = params;
    this.value = `TEST! ${this.params.value}`;
  }

  refresh(params: any): void {
    this.params = params;
    this.value = `TEST! ${this.params.value}`;
  }

  test(): string {
    return this.value;
  }
}
```

My component:
```
import { Component } from "@angular/core";

import { GridOptions } from "ag-grid/main";

import { NameCellComponent } from "./cells/name.component";

@Component({
  selector: "campaigns",
  templateUrl: "./campaigns.component.html"
})

export class CampaignsComponent {
  gridOptions: GridOptions;
  rowData: any[];
  columnDefs: any[];

  constructor(private campaignService: CampaignService) {
    this.gridOptions = <GridOptions>{};
    this.rowData = [];
    this.columnDefs = [
      {
        headerName: "#", width: 30, checkboxSelection: true, suppressSorting: true,
        suppressMenu: true, pinned: true
      },
      {
        headerName: "Name", field: "name", cellRendererFramework: NameCellComponent,
        width: 150, pinned: true
      },
      {
        headerName: "Estimation", field: "estimation",
      },
      {
        headerName: "Action", field: "objective_action",
      },
      {
        headerName: "Cost Per Action", field: "objective_cost_per_action",
      },
      {
        headerName: "Start Time", field: "start_time",
      },
      {
        headerName: "End Time", field: "end_time",
      },
      {
        headerName: "Budget", field: "total_budget",
      },
      {
        headerName: "Bid", field: "bid",
      },
      {
        headerName: "Spend", field: "total_spend",
      },
      {
        headerName: "Impressions", field: "impressions",
      },
      {
        headerName: "Clicks", field: "clicks",
      },
      {
        headerName: "Retweeets", field: "retweets",
      },
      {
        headerName: "Follows", field: "follows",
      }
    ];
  }
}
```

My app.module.ts:

```
import { NgModule }      from "@angular/core";
import { BrowserModule } from "@angular/platform-browser";
import { HttpModule } from "@angular/http";
import { FormsModule, ReactiveFormsModule } from "@angular/forms";
import { RouterModule, Routes } from "@angular/router";

import { Ng2BootstrapModule, ComponentsHelper } from "ng2-bootstrap/ng2-bootstrap";
import { AgGridModule } from "ag-grid-ng2/main";

import { AppComponent }   from "./app.component";

import { CampaignsComponent } from "./components/twitter/campaigns.component";
import { CampaignActionsComponent } from "./components/twitter/campaign-actions.component";
import { CampaignEditComponent } from "./components/twitter/campaign-edit.component";
import { CustomizeTableComponent } from "./components/twitter/customize-table.component";

import { NameCellComponent } from "./components/twitter/cells/name.component";

import { CampaignService } from "./services/twitter/campaign.service";
import { CampaignActionsService } from "./services/twitter/campaign-actions.service";

const appRoutes: Routes = [
  { path: "", component: AppComponent },
  { path: "twitter/campaigns_v2", component: CampaignsComponent },
];

@NgModule({
  imports:      [
    BrowserModule,
    HttpModule,
    FormsModule,
    RouterModule.forRoot(appRoutes),

    Ng2BootstrapModule,
    AgGridModule.withNg2ComponentSupport()
  ],
  providers:    [
    {provide: ComponentsHelper, useClass: ComponentsHelper},
    CampaignService,
    CampaignActionsService
  ],
  declarations: [
    AppComponent,

    CampaignsComponent,
    CampaignActionsComponent,
    CampaignEditComponent,
    CustomizeTableComponent,

    NameCellComponent
  ],
  exports:      [ AppComponent ],
  bootstrap:    [ AppComponent ]
})
export class AppModule { }
```

Am I doing anything wrong when it comes to ng2 cell rendering? Am I missing anything?

Thank you.

**Comment by albertosramires** [comment]
I should mention that the examples for ng2 cell rendering in the official page are broken.

**Comment by albertosramires** [comment]
Fixed by updating to 7.0.0.

**Comment by ceolter** [comment]
thanks for reporting back and closing.
